test/jdk/sun/security/pkcs11/PKCS11Test.java

```diff
@@ -931,6 +919,58 @@ private static String fetchNssLib(Class<?> clazz) {
         return path;
     }
 
+    public abstract void main(Provider p) throws Exception;
+
+    protected boolean skipTest(Provider p) {
+        return false;
+    }
+
+    private void premain(Provider p) throws Exception {
+        if (skipTest(p)) {
+            return;
+        }
+
+        // set a security manager and policy before a test case runs,
+        // and disable them after the test case finished
+        try {
+            if (enableSM) {
+                System.setSecurityManager(new SecurityManager());
+            }
+            long start = System.currentTimeMillis();
+            System.out.printf(
+                    "Running test with provider %s (security manager %s) ...%n",
+                    p.getName(), enableSM ? "enabled" : "disabled");
+            main(p);
+            long stop = System.currentTimeMillis();
+            System.out.println("Completed test with provider " + p.getName() +
+                    " (" + (stop - start) + " ms).");
+        } finally {
+            if (enableSM) {
+                System.setSecurityManager(null);
+            }
+        }
+    }
+
+    // Check support for a curve with a provided Vector of EC support
+    boolean checkSupport(List<ECParameterSpec> supportedEC,
+                         ECParameterSpec curve) {
+        for (ECParameterSpec ec : supportedEC) {
+            if (ec.equals(curve)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    <T> T[] concat(T[] a, T[] b) {
+        if ((b == null) || (b.length == 0)) {
+            return a;
+        }
+        T[] r = Arrays.copyOf(a, a.length + b.length);
+        System.arraycopy(b, 0, r, a.length, b.length);
+        return r;
+    }
+
```

Comment:
I am not sure if 'NO_DEFAULT' expected to be true in setCommonSystemProps() Line:977 or it need to be explicitly reset by each Test where testDefault() expected to be executed,
System.setProperty("NO_DEFAULT", "true"); 

If it is supposed to be true then the testDefault() will always get skipped with the following condition,
Line:270-273 is will return before proceeding to test the default provider,
if ("true".equals(System.getProperty("NO_DEFAULT"))) {
             System.out.println("Skip default provider as test configured with NO_DEFAULT");
             return;
}

Reply:
I see few Tests use "-DNO_DEIMOS=true -DNO_DEFAULT=true" in @run tag when they don't want the specific provider to be used. In that case Line:977 System.setProperty("NO_DEFAULT", "true"); is expected to be "false"

Reply:
There is no change to this logic. Yes, some tests are configured to Skip Deimos and Default run. It is inconsistent as few tests call setCommonSystemProps() and others are setting System property. But, this update has no change to this code.